src/main/java/org/apache/sling/api/resource/external/ExternalizableInputStream.java

```diff
@@ -22,23 +22,25 @@
 import java.net.URI;
 
 import org.jetbrains.annotations.NotNull;
+import org.osgi.annotation.versioning.ProviderType;
 
 /**
  * This interface is normally used to extend an InputStream to indicate that it has a URI form that could
  * be used in place of the InputStream if desired. It is used in situations where the internal of a ResourceProvider
- * wants to offload IO to channels that do not pass through the JVM. The URI that is returned may have restrictions
+ * wants to offload I/O to channels that do not pass through the JVM. The URI that is returned may have restrictions
  * imposed on it requiring it to be used immediately. Do not store the URI for later usage as it will, in most cases,
  * have expired.
- *
+ * Every implementation implementing {@code ExternalizableInputStream} should rely on {@link URIProviders} for getting the URI.
+ * @since 1.0.0 (Sling API Bundle 2.16.4)
  */
+@ProviderType
```

Comment:
This is a breaking change, it will break all existing bundles implementing this type

Reply:
Thanks, reverted in https://github.com/apache/sling-org-apache-sling-api/pull/40/commits/2b8577c3630b5df17a2c6f47975427951cc03822. Still, logically only providers are supposed to implement this IMHO.